aten/src/ATen/detail/PrivateUse1HooksInterface.cpp

```diff
@@ -26,4 +31,20 @@ TORCH_API const at::PrivateUse1HooksInterface& getPrivateUse1Hooks() {
 
 } // namespace detail
 
+struct OpenRegHooksInterface : public at::PrivateUse1HooksInterface {
```

Comment:
Let's move this to torch/csrc in a python-enabled file.
The reason is that it doesn't need to live in ATen and when you will want to call back to python from here (to allow overriding these behaviors from python), you will be able to do so directly without another jump.